core/src/main/scala/chisel3/StrongEnum.scala

```diff
@@ -251,7 +251,25 @@ abstract class EnumType(private[chisel3] val factory: EnumFactory, selfAnnotatin
 
   protected def enumTypeName: String = factory.enumTypeName
 
-  def toPrintable: Printable = FullName(this) // TODO: Find a better pretty printer
+  def toPrintable: Printable = {
+    implicit val sourceInfo = UnlocatableSourceInfo
+    implicit val compileOptions = ExplicitCompileOptions.Strict
+    val allNames = factory.allNames.zip(factory.all)
+    val nameSize = allNames.map(_._1.length).max
+    val allNamesPadded = allNames.map { case (name, value) => name.padTo(nameSize, ' ') -> value }
+
+    val result = Wire(Vec(nameSize, UInt(8.W))).suggestName("EnumRecord")
```

Comment:
```suggestion
    val result = Wire(Vec(nameSize, UInt(8.W))).suggestName(s"_${enumTypeName}Printable")
```
I think something that makes it a little more clear where this comes from since we don't give a source locator.